crates/typst/src/visualize/gradient.rs

```diff
@@ -1249,7 +1249,11 @@ fn sample_stops(stops: &[(Color, Ratio)], mixing_space: ColorSpace, t: f64) -> C
 
     let (col_0, pos_0) = stops[low - 1];
     let (col_1, pos_1) = stops[low];
-    let t = (t - pos_0.get()) / (pos_1.get() - pos_0.get());
+    let mut delta = pos_1.get() - pos_0.get();
+    if delta == 0.0 {
+        delta = 1.0;
+    }
+    let t = (t - pos_0.get()) / delta;
```

Comment:
We can also deduplicate stops beforehand, but I think we should in any case clamp `t` to `0.0..=1.0` here. You never know what's gonna happen with floats.